### PR TITLE
add is_keyer macro

### PR DESCRIPTION
**Pull request by Andersama**
Adds a macro to try to detect a key-like structure.

TODO: key-like structures are any integral, bitset, boolean, float, double, tuple of keys or array of keys.

Also changes `is_comparer` to take unsigned integers.

### CODE REVIEW

File: lib/std/sort/sort.c3 [hunk 1]
```diff
@@ -13,7 +13,7 @@ macro usz @len_from_list(&list)
 macro bool @is_comparer(#cmp, #list)
 {
 	var $Type = $typeof(#cmp);
-	$if $and($Type.kindof == FUNC, $Type.returns.kindof == SIGNED_INT):
+	$if $and($Type.kindof == FUNC, $or($Type.returns.kindof == SIGNED_INT, $Type.returns.kindof == UNSIGNED_INT)):
```
Comment (lerno) [inline lib/std/sort/sort.c3:16]: This is incorrect, a comparator function should return -1, 0, 1. And unsigned int cannot ever return -1

**Comment by lerno** [comment]
What should is_keyer do?

File: lib/std/sort/sort.c3 (hunk 1 above)
Comment (Andersama) [reply lib/std/sort/sort.c3:16]: Ah, ok, I didn't realize that the macro was testing for three way comparisons.

File: lib/std/sort/sort.c3 [hunk 2]
```diff
@@ -35,4 +35,44 @@ macro bool @is_comparer(#cmp, #list)
...
-}
\ No newline at end of file
+}
+
+macro bool @is_key(#item)
+{
+	var $key_type = $typeof(#item);
+	$switch
+		$case $key_type == @typeid(char):
+			return true;
```
Comment (lerno) [inline lib/std/sort/sort.c3:45]: Two alternatives:
1. Use kindof, or just `.is_int()`:  `$typeof(#item).is_int()` returns true for all integer types.
2. If you want to do a type switch, it easy:

```c
$switch $typeof(#item)
$case char: return true;
$case ushort: return true;
```

**Comment by lerno** [comment]
Added as @is_cmp_key_fn in `dev`

**Pull request closed without merging**